tests/returntypes/transforms/test_mitigate_new.py

```diff
@@ -0,0 +1,570 @@
+        noise_strength = 0.05
+
+        dev_noise_free = qml.device("default.mixed", wires=2)
+        dev = qml.transforms.insert(qml.AmplitudeDamping, noise_strength)(dev_noise_free)
+
+        n_wires = 2
+        n_layers = 2
+
+        shapes = qml.SimplifiedTwoDesign.shape(n_wires, n_layers)
+        np.random.seed(0)
+        w1, w2 = [np.random.random(s) for s in shapes]
+
+        @qml.transforms.mitigate_with_zne([1, 2, 3], fold_global, richardson_extrapolate)
+        @qml.qnode(dev)
+        def mitigated_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
+
+        @qml.qnode(dev_noise_free)
+        def ideal_circuit(w1, w2):
+            qml.SimplifiedTwoDesign(w1, w2, wires=range(2))
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
+
+        res_mitigated = mitigated_circuit(w1, w2)
+        res_ideal = qml.math.stack(ideal_circuit(w1, w2))
+
+        assert res_mitigated.shape == res_ideal.shape
+        assert not np.allclose(res_mitigated, res_ideal)
+
+    def test_reps_per_factor_not_1(self, mocker):
+        """Tests if mitigation proceeds as expected when reps_per_factor is not 1 (default)"""
+        # qml.disable_return()
+
+        scale_factors = [1, 2, -4]
+        spy_fold = mocker.spy(self, "folding")
+        spy_extrapolate = mocker.spy(self, "extrapolate")
+        tapes, fn = mitigate_with_zne(
+            tape, scale_factors, self.folding, self.extrapolate, reps_per_factor=2
+        )
+        random_results = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6]
+
+        args = spy_fold.call_args_list
+        for i in range(6):
+            same_tape(args[i][0][0], tape_base)
+        assert [args[i][0][1] for i in range(6)] == [1, 1, 2, 2, -4, -4]
+
+        fn(random_results)
+
+        args = spy_extrapolate.call_args
+
+        assert args[0][0] == scale_factors
+        assert np.allclose(args[0][1], np.mean(np.reshape(random_results, (3, 2)), axis=1))
+
+
+@pytest.fixture
+def skip_if_no_mitiq_support():
+    """Fixture to skip if minimum version of mitiq is not available"""
+    try:
+        import mitiq
+
+        v = version.parse(mitiq.__version__)
+        t = version.parse("0.11.0")
+        if v.major < t.major and v.minor < t.minor:
+            pytest.skip("Mitiq version too low")
+    except ImportError:
+        pytest.skip("Mitiq not available")
+
+
+@pytest.fixture
+def skip_if_no_pl_qiskit_support():
+    """Fixture to skip if pennylane_qiskit is not available"""
+    pytest.importorskip("pennylane_qiskit")
+
+
+@pytest.mark.usefixtures("skip_if_no_pl_qiskit_support")
+@pytest.mark.usefixtures("skip_if_no_mitiq_support")
+class TestMitiqIntegration:
+    """Tests if the mitigate_with_zne transform is compatible with using mitiq as a backend"""
+
+    def test_multiple_returns(self):
+        """Tests if the expected shape is returned when mitigating a circuit with two returns"""
```

Comment:
Updated this test to stack the results of `ideal_circuit`.

Reply:
Why was that required?

Reply:
It was required because `ideal_circuit` is a QNode with multiple measurements. With the old return types this would already return a stacked array, but with the new return types we need to stack manually.

Reply:
Right, but why wasn't the output of `mitigated_circuit(w1, w2)` stacked then? :thinking:

Reply:
The `processing_fn` of the transform does the stacking internally. I think this is required for the extrapolation to work.